extensions/ffghdfh/indexedDB.js

```diff
@@ -0,0 +1,221 @@
+// Name: Indexed Database
+// ID: ffghdfhIndexedDB
+// Description: Lets you use an IndexedDB database, allowing for you to store more than 5MB of data locally.
+// License: LGPL-3.0
+(function(Scratch) {
+    "use strict";
+  if (!Scratch.extensions.unsandboxed) {
+    throw new Error("Indexed Database must be run unsandboxed");
+  }
+
+class IndexedDBExtension {
+  constructor() {
+    this.dbName = 'MyScratchDB'; // Default database name
+    this.db = null;
+  }
+  
+    // Set the database name
+    _setDBName(name) {
+      this.dbName = name;
+    }
+  
+    // Initialize the IndexedDB
+    _initDB() {
+      return new Promise((resolve, reject) => {
+        const request = indexedDB.open(this.dbName, 1);
+  
+        request.onerror = (event) => {
+          console.error('Database error:', event.target.error);
+          reject(event.target.error);
+        };
+  
+        request.onsuccess = (event) => {
+          this.db = event.target.result;
+          resolve(this.db);
+        };
+  
+        request.onupgradeneeded = (event) => {
+          const db = event.target.result;
+          db.createObjectStore('data', { keyPath: 'key' });
+        };
+      });
+    }
+  
+    // Set value in the IndexedDB
+    _setValue(key, value) {
+      return this._initDB().then((db) => {
+        const transaction = db.transaction(['data'], 'readwrite');
+        const store = transaction.objectStore('data');
+        const request = store.put({ key, value });
+  
+        return new Promise((resolve, reject) => {
+          request.onsuccess = () => resolve();
+          request.onerror = () => reject(request.error);
+        });
+      });
+    }
+  
+    // Get value from the IndexedDB
+    _getValue(key) {
+      return this._initDB().then((db) => {
+        const transaction = db.transaction(['data'], 'readonly');
+        const store = transaction.objectStore('data');
+        const request = store.get(key);
+  
+        return new Promise((resolve, reject) => {
+          request.onsuccess = () => resolve(request.result ? request.result.value : null);
+          request.onerror = () => reject(request.error);
```

Comment:
Slight issue, in unsandboxed extensions [blocks throwing errors can break stuff](https://docs.turbowarp.org/development/extensions/unsandboxed#increased-power-brings-increased-responsibility). You should probably just resolve with an empty string (for reporters) or undefined (for commands) in error cases. (This also applies to all other times `reject()` is used in blocks.)

Reply:
Oh gee, didn't know that! Thanks for letting me know! I'm rally new to github though, so how do I modify my changes 'n stuff?